app/models/alert.rb

```diff
@@ -7,6 +7,14 @@ class Alert < ApplicationRecord
   self.default_sort_direction = :desc
   self.allowed_search_scopes = %w[cinstance_id account_id timestamp level]
 
+  self.background_deletion_method = :delete_truly
+  # commenting out this here just to make a note that to me it seems like state machine overrides
+  #   the #delete AR method but what background deletion actually expects is indeed the AR #delete method
+  # self.background_deletion_scope_name = :non_deleted
+  #
+  # scope :non_deleted, -> { where.not(state: :deleted) }
+  alias delete_truly delete # keep reference to the original AR #delete method to use for background deletion
```

Comment:
Rather than aliasing `delete` as `delete_truly`, I would really rather just change the method that changes the alert state:
```
    event :mark_as_deleted do
      transition [:unread, :read] => :deleted
    end
```

And then, in both controllers that are currently calling `alert.delete!` change this invocation to `alert.mark_as_deleted!`. I think that would be much more clear.

Indeed, I think that the alerts not being deleted when the corresponding cinstance (or the account owning it) is deleted is just an oversight.
Actually, the fact that the alerts are just marked as deleted and not actually deleted on UI actions is also questionable, I think. But I guess this is because there are two "consumers" of the alerts - the provider and the buyer.

Reply:
Agreed, but I'd like it to be a separate PR.

Reply:
How about first doing this refactoring in a separate PR, confirming that nothing is broken (as per the tests), and then rebase this PR on top of it?

I think it's more confusing to add these aliases, and then to remove them...

Reply:
Not really enthusiastic. I'll check if it is easy and give up if not. ok?

Reply:
Seemed fairly easy to me, like 3 small changes.

Reply:
Also, I think it would be nice to rebase anyway, because the current branch seems to be based on Rails 6.1